On_ping bug

Small bug in the discovery community, the peers who send you a ping aren't neccesarily a walkcandidate. Use the candidate of the actual taste buddy.

```
  File "/home/jenkins/workspace/Test_community_walking/dispersy/dispersy.py", line 1702, in _update
    messages[0].handle_callback(messages)
  File "/home/jenkins/workspace/Test_community_walking/dispersy/discovery/community.py", line 663, in on_ping
    self.reset_taste_buddy(message.candidate)
  File "/home/jenkins/workspace/Test_community_walking/dispersy/discovery/community.py", line 323, in reset_taste_buddy
    self.peer_cache.add_or_update_peer(candidate)
  File "/home/jenkins/workspace/Test_community_walking/dispersy/discovery/community.py", line 744, in add_or_update_peer
    assert isinstance(wcandidate, WalkCandidate), type(wcandidate)
exceptions.AssertionError: <class 'dispersy.candidate.Candidate'>
```

--- a/discovery/community.py
+++ b/discovery/community.py
@@ -316,7 +316,7 @@ class DiscoveryCommunity(Community):
             if tb == candidate:
                 tb.timestamp = time()
                 if tb.should_cache():
-                    self.peer_cache.add_or_update_peer(candidate)
+                    self.peer_cache.add_or_update_peer(tb.candidate)
                 break
 
     def remove_taste_buddy(self, candidate):

